src/core/analytics.js

```diff
@@ -868,19 +916,17 @@ class Analytics {
         );
       }
 
-      // convert integrations object to server identified names, kind of hack now!
-      transformToServerNames(rudderElement.message.integrations);
-      rudderElement.message.integrations = getMergedClientSuppliedIntegrations(
-        this.integrationsData,
-        clientSuppliedIntegrations,
-      );
-
-      // self analytics process, send to rudder
-      this.eventRepository.enqueue(rudderElement, type);
-
-      // logger.debug(`${type} is called `)
-      if (callback) {
-        callback(rudderElement);
+      // Holding the cloud mode events based on flag and integrations load check
+      if (!this.bufferDataPlaneEventsUntilReady || this.clientIntegrationObjects) {
+        this.processCloudModeEvents(type, rudderElement, callback, clientSuppliedIntegrations);
+      } else {
+        const cloudModeEvents = this.store.get('rs_events');
```

Comment:
How are we going to handle unprocessed events from previous SDK sessions?

How will we handle when two web page tabs are active simultaneously?

Reply:
Are these concerns addressed?

Reply:
I would go like:
- add another queue in the eventRepository
- create a new Queue class similar to XHRQueue, like PreProcessQueue that is just invoking a callback in processQueueElement method (instead of sending requests)
- enqueue all the buffered events in this new store queue (this will take care of multiple tabs etc, same way we handle our event queue now
- once we are ready to send them process thus queue and call this.eventRepository.enqueue(rudderElement, type); for each

This will mean thta we handle and store our queues in ocalstorage with same logis and dependencies

Reply:
Yes. @mihir-4116, this is what we discussed yesterday.